translator/test-steps/src/main/java/org/eclipse/kapua/translator/test/steps/TranslatorSteps.java

```diff
@@ -0,0 +1,363 @@
+        translatorResponseMqttKura = new TranslatorResponseMqttKura();
+        translatorDataKuraMqtt = new TranslatorDataKuraMqtt();
+        translatorDataJmsKura = new TranslatorDataJmsKura();
+        translatorDataKuraJms = new TranslatorDataKuraJms();
+
+        if (isUnitTest()) {
+            // Create KapuaSession using KapuaSecurtiyUtils and kapua-sys user as logged in user.
+            // All operations on database are performed using system user.
+            // Only for unit tests. Integration tests assume that a real logon is performed.
+            KapuaSession kapuaSession = new KapuaSession(null, SYS_SCOPE_ID, SYS_USER_ID);
+            KapuaSecurityUtils.setSession(kapuaSession);
+        }
+
+        // Setup JAXB context
+        XmlUtil.setContextProvider(new TestJAXBContextProvider());
+    }
+
+    @After
+    public void afterScenario() {
+
+        // Clean up the database
+        try {
+            logger.info("Logging out in cleanup");
+            if (isIntegrationTest()) {
+                database.deleteAll();
+                SecurityUtils.getSubject().logout();
+            } else {
+                database.dropAll();
+                database.close();
+            }
+            KapuaSecurityUtils.clearSession();
+        } catch (Exception e) {
+            logger.error("Failed to log out in @After", e);
+        }
+    }
+
+    @Given("^I try to translate from \"([^\"]*)\" to \"([^\"]*)\"$")
+    public void iFindTranslator(String from, String to) throws Exception {
+        Class fromClass;
+        Class toClass;
+        try {
+            if (!from.equals("") && !to.equals("")) {
+                fromClass = Class.forName(from);
+                toClass = Class.forName(to);
+            } else {
+                fromClass = null;
+                toClass = null;
+            }
+            Translator translator = Translator.getTranslatorFor(exampleTranslator.getClass(fromClass), exampleTranslator.getClass(toClass));
+            stepData.put("Translator", translator);
+        } catch (Exception ex) {
+            verifyException(ex);
+        }
+    }
+
+    @Then("^Translator \"([^\"]*)\" is found$")
+    public void translatorIsFound(String translatorName) {
+        Translator translator = (Translator) stepData.get("Translator");
+        assertEquals(translatorName, translator.getClass().getSimpleName());
+    }
+
+    @Given("^I create mqtt message with (?:valid|invalid|empty) payload \"([^\"]*)\" and (?:valid|invalid) topic \"([^\"]*)\"$")
+    public void iTryToProvideInvalidTopic(String payload, String topic) {
+
+        Date date = new Date();
+        MqttTopic mqttTopic = new MqttTopic(topic);
+
+        KuraPayload kuraPayload = new KuraPayload();
+        if (payload.equals("invalidPayload") || payload.equals("")) {
+            kuraPayload.setBody(payload.getBytes());
+        } else {
+            kuraPayload.getMetrics().put(payload, 200);
+        }
+
```

Comment:
insert try/catch block, the "stepdata.put" part can be left ouside the try/catch.